### ISSUE TITLE
Revise "Dominant Color" styling and inline styling

### ISSUE THREAD

**Issue by jamesozzie** [WordPress/performance#577]
## Bug Description

As [one user in the support forums](https://wordpress.org/support/topic/dominant-color-option-enabled-it-overwrites-style-only-on-the-frontend/#post-16172789) reported, the plugin is adding the below inline styling to images if the "_Dominants Color_" option is enabled. 
`style="--dominant-color: #332f2b;"`

One of the problems with this, is that it is overwriting any existing inline styling associated with any images. An example of this can be demonstrated in the below recording, where you'll find that my own custom inline styling is removed, in place of the PL inline styling. As soon as I disabled the "_Dominant Color_" option my custom inline styling is applied once more.  

https://user-images.githubusercontent.com/41326532/200372452-ce2b835e-2021-4167-917c-2f37ca30257f.mp4

To resolve this, consider moving away from inline styling. Printing the styling to the `wp_head` as opposed to within the `img` element also resolves this error. Tested with the below, which doesn't conflict with the "Dominant color" configuration styling:

```
function inline_css() {
  echo "<style>
  .wp-block-image img{opacity: 0.5;}
  </style>";
}
add_action( 'wp_head', 'inline_css', 0 );
```

Note also that there is [a GitHub issue](https://github.com/WordPress/gutenberg/issues/23893) in relation to not allowing themes or plugins to insert inline styling. 

## Steps to reproduce

1. Ensure you have the Performance Lab plugin active but the "Dominant Color" option disabled
2. Create a new post and switch to the code editor. 
3. Add some inline styling to an image or paste the code snippet highlighted below (_Example snippet_) into your content. Check that the styling does work as intended by visiting the front end of your site and viewing that page
4. Enable the "Dominent Color" option
5. Revisit the page on your site where you added your own inline styling
6. Refresh the page and notice that your own custom inline styling is replaced

<Details>
<summary>Example snippet</summary>

```
<!-- wp:image {"id":6632,"sizeSlug":"full","linkDestination":"none"} -->
<figure class="wp-block-image size-full"><img src="https://www.gstatic.com/webp/gallery/1.webp" alt="" class="wp-image-6632"/ style="opacity: 0.5;border:5px solid #000;border-radius:20px"></figure>
<!-- /wp:image -->
```

</details>

**Comment by mxbclang** [comment]
@pbearne @spacedmonkey Tagging you in on this report from a user from the support forum. Can you please take a look?

**Comment by felixarntz** [comment]
Fixed via #578 and #582
